sdk/trace/provider.go

```diff
@@ -109,32 +109,52 @@ func (p *Provider) Tracer(name string, opts ...apitrace.TracerOption) apitrace.T
 func (p *Provider) RegisterSpanProcessor(s SpanProcessor) {
 	p.mu.Lock()
 	defer p.mu.Unlock()
-	new := make(spanProcessorMap)
-	if old, ok := p.spanProcessors.Load().(spanProcessorMap); ok {
-		for k, v := range old {
-			new[k] = v
+	new := spanProcessorStates{}
+	if old, ok := p.spanProcessors.Load().(spanProcessorStates); ok {
+		for _, v := range old {
+			new = append(new, v)
 		}
 	}
```

Comment:
```suggestion
	if old, ok := p.spanProcessors.Load().(spanProcessorStates); ok {
		copy(new, old)
	}
```

Should this use `copy(new, old)` instead? That might fix the build failure.